### PR TITLE
Fix ZeroFillRShift tokenized as Gt

### LINKED ISSUE #751: ZeroFillRShift tokenized as Gt
**Describe the bug**

There is a scenario where a zero fill right shift (`>>>`) gets tokenized as greater than.

**Input code**

```ts
t ? -(v >>> 1) : v >>> 1
```

**Tokens**

Notice the second last line:

```
TokenAndSpan { token: Word(t), had_line_break: true, span: Span { lo: BytePos(0), hi: BytePos(1), ctxt: #0 } },
TokenAndSpan { token: QuestionMark, had_line_break: false, span: Span { lo: BytePos(2), hi: BytePos(3), ctxt: #0 } },
TokenAndSpan { token: BinOp(Sub), had_line_break: false, span: Span { lo: BytePos(4), hi: BytePos(5), ctxt: #0 } },
TokenAndSpan { token: LParen, had_line_break: false, span: Span { lo: BytePos(5), hi: BytePos(6), ctxt: #0 } },
TokenAndSpan { token: Word(v), had_line_break: false, span: Span { lo: BytePos(6), hi: BytePos(7), ctxt: #0 } },
TokenAndSpan { token: BinOp(ZeroFillRShift), had_line_break: false, span: Span { lo: BytePos(8), hi: BytePos(11), ctxt: #0 } },
TokenAndSpan { token: Num(1.0), had_line_break: false, span: Span { lo: BytePos(12), hi: BytePos(13), ctxt: #0 } },
TokenAndSpan { token: RParen, had_line_break: false, span: Span { lo: BytePos(13), hi: BytePos(14), ctxt: #0 } },
TokenAndSpan { token: Colon, had_line_break: false, span: Span { lo: BytePos(15), hi: BytePos(16), ctxt: #0 } },
TokenAndSpan { token: Word(v), had_line_break: false, span: Span { lo: BytePos(17), hi: BytePos(18), ctxt: #0 } },
TokenAndSpan { token: BinOp(Gt), had_line_break: false, span: Span { lo: BytePos(19), hi: BytePos(20), ctxt: #0 } },
TokenAndSpan { token: Num(1.0), had_line_break: false, span: Span { lo: BytePos(23), hi: BytePos(24), ctxt: #0 } }
```

**Version**
swc_ecma_parser: 0.21.9

### PR DESCRIPTION
**Pull request by dsherret**
Closes #751.

### CODE REVIEW

**Comment by kdy1** [comment]
I'll check it (and fix if required) tonight

**Comment by dsherret** [comment]
Oops... didn't realize this broke JSX elements... hmmm... I'll also try to look into this more soon.

**Comment by dsherret** [comment]
@kdy1 this should be actually fixed now.

File: ecmascript/parser/src/parser/input.rs [hunk 1]
```diff
@@ -118,9 +118,13 @@ impl<I: Tokens> Iterator for Capturing<I> {
         match next {
             Some(ts) => {
                 let mut v = self.captured.borrow_mut();
-                if let Some(last) = v.last() {
+
+                // remove tokens that could change due to backtracing
+                while let Some(last) = v.last() {
                     if last.span.lo >= ts.span.lo {
-                        return Some(ts);
+                        v.pop();
```
Comment (dsherret) [inline ecmascript/parser/src/parser/input.rs:125]: I'm not sure whether this is faster or removing a slice from the vector (which I'm not sure how to do in Rust).

Comment (kdy1) [reply ecmascript/parser/src/parser/input.rs:125]: I'm not sure, but I think compiler will optimize it anyway

**Comment by kdy1** [comment]
Thank you very much!!

**Pull request merged**